contracts/Committee.sol

```diff
@@ -0,0 +1,548 @@
+			return (false, 0);
+		}
+
+		(bool found, uint atPos) = findTimedOutStandby();
+		if (found) {
+			return (true, atPos);
+		}
+
+		uint standbyCount = topology.length - committeeSize;
+		if (standbyCount < maxStandbys) {
+			return (true, topology.length);
+		}
+
+		(bool foundLowest, uint lowestRankedStandbyPos, uint lowestRankedStandbyWeight) = findLowestWeightStandby();
+		if (foundLowest && lowestRankedStandbyWeight < getValidatorWeight(validator)) {
+			return (true, lowestRankedStandbyPos);
+		}
+
+		return (false, 0);
+	}
+
+	function isAboveCommitteeEntryThreshold(address validator) private view returns (bool) {
+		return _compareValidatorsByData(minimumAddress, minimumWeight, true, validator, getValidatorWeight(validator), true) < 1;
+	}
+
+	function _isQualifiedForCommitteeByRank(address validator) private view returns (bool qualified, uint entryPos) {
+		// this assumes maxTopologySize > maxCommitteeSize, otherwise a non ready-for-committee validator may override one that is ready.
+		if (!isReadyForCommittee(validator) || isReadyToSyncStale(validator)) {
+			return (false, 0);
+		}
+
+		if (committeeSize >= minCommitteeSize && !isAboveCommitteeEntryThreshold(validator)) {
+			return (false, 0);
+		}
+
+		if (committeeSize < maxCommitteeSize || _compareValidators(topology[committeeSize - 1], validator) < 0) {
+			return (true, topology.length);
+		}
+
+		return (false, 0);
+	}
+
+	function findTimedOutStandby() private view returns (bool found, uint pos) {
+		pos = oldestReadyToSyncStandbyPos;
+		found = pos >= committeeSize && pos < topology.length && isReadyToSyncStale(topology[pos]);
+	}
+
+	function findLowestWeightStandby() private view returns (bool found, uint pos, uint weight) {
+		if (topology.length == committeeSize) {
+			return (false, 0, 0);
+		}
+
+		address v1 = topology[topology.length - 1];
+		uint256 v1Weight = getValidatorWeight(v1);
+		if (readyForCommitteeCount <= committeeSize) {
+			return (true, topology.length - 1, v1Weight);
+		}
+
+		address v2 = topology[readyForCommitteeCount - 1];
+		uint256 v2Weight = getValidatorWeight(v2);
+		if (v2Weight < v1Weight) {
+			return (true, readyForCommitteeCount - 1, v2Weight);
+		}
+
+		return (true, topology.length - 1, v1Weight);
+	}
+
+	function _findInTopology(address v) private view returns (uint, bool) {
+		uint l =  topology.length;
+		for (uint i=0; i < l; i++) {
+			if (topology[i] == v) {
+				return (i, true);
+			}
+		}
+		return (0, false);
+	}
+
+	function _removeFromTopology(address addr) private returns (bool committeeChanged, bool standbysChanged) {
+		standbysChanged = false;
+		committeeChanged = false;
+		(uint pos, bool inTopology) = _findInTopology(addr);
+		if (inTopology) {
+			_evict(pos);
+
+			(uint prevCommitteeSize, uint currentCommitteeSize) = _onTopologyModification();
+
+			if (prevCommitteeSize != currentCommitteeSize || pos < currentCommitteeSize) { // was in committee
+				committeeChanged = true;
+				_notifyCommitteeChanged();
```

Comment:
BUG - missing check for standbys change as a result of a standby joining the committee